src/gromacs/nblib/box.cpp

```diff
@@ -72,14 +74,16 @@ Box::Box(real x, real y, real z)
     box_(XX, XX) = x;
     box_(YY, YY) = y;
     box_(ZZ, ZZ) = z;
+
+    gmx::fillLegacyMatrix(box_, legacyMatrix);
 }
 
-Box::Matrix Box::matrix()
+Box::Matrix Box::getMatrix()
```

Comment:
The keyword `matrix` as the typedef in GROMACS conflicts with the function name `matrix()`. Hence, had to rename it.

On another note: Does NB-LIB's user facing code have to adhere to GROMACS's naming guidelines? I understand that the translation layer should. Like this case is purely due to conflicts with GROMACS's internal naming.

Reply:
There is no collision between matrix and matrix() so the change to getMatrix() at this point would be purely aesthetic. I think we should adhere to gromacs coding style even in the public interface unless there is a compelling reason not to, because then we can just link to that coding style document in our own docs. Also, I find getMatrix() needlessly verbose. Just typing matrix() is clear enough.

Reply:
Collision resolved using `using`. Hope the C array usage goes away soon and we use the modern `Matrix` instead.